main/inc/lib/sessionmanager.lib.php

```diff
@@ -5429,4 +5428,266 @@ public static function getCoursesListByCourseCoach($coachId)
         ));
     }
 
+    /**
+     * Returns the list of session (name, short description, start date, end date) from category.
+     * The short description is an extra field value
+     * @param int $categoryId
+     * @param string $target
+     * @return mixed
+     */
+    public static function getSessionBriefListByCategory($categoryId, $target) {
+        // Init variables
+        $categoryId = (int) $categoryId;
+        $sessionList = array();
+        // Check if categoryId is valid
+        if ($categoryId > 0) {
+            $target = Database::escape_string($target);
+            $sTable = Database::get_main_table(TABLE_MAIN_SESSION);
+            $sfTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD);
+            $sfvTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD_VALUES);
+            // Join session field and session field values tables
+            $joinTable = $sfTable . ' sf INNER JOIN ' . $sfvTable . ' sfv ON sf.id = sfv.field_id';
+            $fieldsArray = array(
+                'short_description', 'mode', 'duration', 'vacancies', 'brochure', 'target', 'schedule'
```

Comment:
Duration (integer field, number of days) is now part of the session table in 1.10.x, so it should move *out* of extra fields and into session.